src/openrct2/actions/RideEntranceExitRemoveAction.hpp

```diff
@@ -0,0 +1,155 @@
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "../ride/Ride.h"
+#include "../ride/Station.h"
+#include "../world/Entrance.h"
+#include "GameAction.h"
+
+DEFINE_GAME_ACTION(RideEntranceExitRemoveAction, GAME_COMMAND_REMOVE_RIDE_ENTRANCE_OR_EXIT, GameActionResult)
+{
+private:
+    CoordsXY _loc;
+    NetworkRideId_t _rideIndex;
+    uint8_t _stationNum;
+    bool _isExit;
+
+public:
+    RideEntranceExitRemoveAction() = default;
+
+    RideEntranceExitRemoveAction(CoordsXY loc, ride_id_t rideIndex, uint8_t stationNum, bool isExit)
+        : _loc(loc)
+        , _rideIndex(rideIndex)
+        , _stationNum(stationNum)
+        , _isExit(isExit)
+    {
+    }
+
+    uint16_t GetActionFlags() const override
+    {
+        return GameAction::GetActionFlags();
+    }
+
+    void Serialise(DataSerialiser & stream) override
+    {
+        GameAction::Serialise(stream);
+
+        stream << DS_TAG(_loc) << DS_TAG(_rideIndex) << DS_TAG(_stationNum) << DS_TAG(_isExit);
+    }
+
+    GameActionResult::Ptr Query() const override
+    {
+        if (_rideIndex >= MAX_RIDES || _rideIndex == RIDE_ID_NULL)
+        {
+            log_warning("Invalid game command for ride %u", uint32_t(_rideIndex));
+            return std::make_unique<GameActionResult>(GA_ERROR::INVALID_PARAMETERS, STR_NONE);
+        }
+
+        Ride* ride = get_ride(_rideIndex);
+        if (ride == nullptr || ride->type == RIDE_TYPE_NULL)
+        {
+            log_warning("Invalid ride id %u for entrance/exit removal", _rideIndex);
+            return std::make_unique<GameActionResult>(GA_ERROR::INVALID_PARAMETERS, STR_NONE);
+        }
+
+        if (ride->status != RIDE_STATUS_CLOSED)
+        {
+            gGameCommandErrorText = STR_MUST_BE_CLOSED_FIRST;
+            return MakeResult(GA_ERROR::INVALID_PARAMETERS, STR_MUST_BE_CLOSED_FIRST);
+        }
+
+        if (ride->lifecycle_flags & RIDE_LIFECYCLE_INDESTRUCTIBLE_TRACK)
+        {
+            return MakeResult(GA_ERROR::INVALID_PARAMETERS, STR_NOT_ALLOWED_TO_MODIFY_STATION);
+        }
+
+        return MakeResult();
+    }
+
+    GameActionResult::Ptr Execute() const override
+    {
+        Ride* ride = get_ride(_rideIndex);
+        if (ride == nullptr || ride->type == RIDE_TYPE_NULL)
+        {
+            log_warning("Invalid ride id %u for entrance/exit removal", _rideIndex);
+            return std::make_unique<GameActionResult>(GA_ERROR::INVALID_PARAMETERS, STR_NONE);
+        }
+
+        ride_clear_for_construction(ride);
+        ride_remove_peeps(ride);
+        invalidate_test_results(ride);
+
+        bool found = false;
+        TileElement* tileElement = map_get_first_element_at(_loc.x / 32, _loc.y / 32);
+
+        do
```

Comment:
This part is missing in Query

Reply:
This was not originally in the query but it does make sense to add it. Will change it.